File: sources/path-service/src/main/java/fr/unice/polytech/si5/al/e/ControlTravel.java
package fr.unice.polytech.si5.al.e;

import fr.unice.polytech.si5.al.e.model.Customer;
import fr.unice.polytech.si5.al.e.model.Item;
import fr.unice.polytech.si5.al.e.model.Travel;
import fr.unice.polytech.si5.al.e.model.exceptions.NoSuchCustomerIdException;

import javax.ejb.Local;
import java.util.List;

@Local
public interface ControlTravel {
    Travel createTravel(String customerName, String departure, String destination);

    Travel addItemToTravel(Item item, String travelId);

    List<Travel> findTravel(String departure, String destination);

    List<Travel> findTravel();

    Travel findTravelById(String travelId);

    Travel chooseTravel(String transporterName, String travelId);

    void finishTravel(String travelId);

    Customer getCustomerById(int id) throws NoSuchCustomerIdException;
}

File: blablamoveService/typescript/main/src/path-service/ControlTravel.ts
import { Travel } from '../entity/Travel';
import { Item } from "../entity/Item";
import { Customer } from '../entity/Customer';

export interface ControlTravels {
    createTravel(
        customerName: string,
        departure: string,
        destination: string
    ): Travel;

    addItemToTravel(item: Item, travelId: string): Travel;

    findTravel(departure: string, destination: string): Travel[];

    findTravelById(travelId: string): Travel;

    chooseTravel(transporterName: string, travelId: string): Travel;

    finishTravel(travelId: string): void;

    getCustomerById(id: string): Customer;
}